code/script.js

```diff
@@ -246,98 +280,132 @@ const generateBoard = () => {
-
-  if (category === 'hair color') {
+  const value = questions.options[questions.selectedIndex].value
+  if (category === 'Färg') {
     currentQuestion = {
-      attribute: 'hairColor',
-      // value: ,
-      // 👆 add the value from the input here
-      category: category,
+      attribute: 'color',
+      value,
+      category,
     }
-  } else if (category === 'eye color') {
-    // Set this up your self
-  } else if (category === 'accessories') {
+  } else if (category === 'Antal ben') {
     currentQuestion = {
-      //attribute: ,
-      // 👆 this is the property of the booleans such as smoke, glasses and hat. add the value from the input here
-      value: true, // we're asking if this person wears a hat for exaple, so always true in the question.
-      category: category,
+      attribute: 'numberOfLegs',
+      value,
+      category
+    }
+  } else if (category === 'Vad den gör') {
+    currentQuestion = {
+      attribute: value,
+      value: true, 
+      category
     }
-  } else if (category === 'other') {
-    // Set this up your self (should be same structure as above)
   }
 }
 
 // This function should be invoked when you click on 'Find Out'.
 const checkQuestion = () => {
-  // Compare the currentQuestion with the secret person.
-  // See if we should keep or remove people based on that
-  // Then invoke filterCharacters
+  numberOfCounts()
+  selectQuestion()
+  const secretValue = secret[currentQuestion.attribute]
+  if (secretValue === currentQuestion.value) {
+    filterCharacters(true)
+  } else {
+    filterCharacters(false)
+  }
+}
+
+const numberOfCounts = () => {
+  counter += 1
+  document.getElementById("counter").innerHTML = counter
 }
 
+
 // It'll filter the characters array and redraw the game board.
 const filterCharacters = (keep) => {
-  // Show the correct alert message for different categories
-  if (group === 'accessories') {
-    if (keep) {
-      alert(
-        `Yes, the person wears ${attribute}! Keep all that wears ${attribute}`
-      )
-    } else {
-      alert(
-        `No, the person doesn't wear ${attribute}! Remove all that wears ${attribute}`
-      )
-    }
-  } else if (group === 'other') {
-    // Similar to the one above
-  } else {
+  const group = currentQuestion.category
+  const value = currentQuestion.value
+  const attribute = currentQuestion.attribute
+  message = ""
+  if (group === 'Färg') {
     if (keep) {
-      // alert popup that says something like: "Yes, the person has yellow hair! Keep all persons with yellow hair"
+      message = `<div class="answer-yes">STÄMMER</div><strong>Djuret har ${value} päls!</strong> <div>Behåller alla djur med ${value} päls.</div>`
     } else {
-      // alert popup that says something like: "NO, the person doesnt have yellow hair! Remove all persons with yellow hair"
+      message = `<div class="answer-no">TYVÄRR</div><strong>Djuret har inte ${value} päls!</strong> <div>Tar bort alla djur med ${value} päls...</div>`
```

Comment:
Very nice feature to make the response into a message displayed in the question section, instead of being a pop up alert! When the player is selecting the animal characteristic of "har ingen päls", the message becomes a bit peculiar. The message could perhaps somehow be revised into "Tyvärr, djuret har ingen päls! Tar bort alla djur utan päls" instead of "Tyvärr, djuret har inte ingen päls! Tar bort alla djur med ingen päls..."